ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -228,7 +228,26 @@ protected BlockCreationResult createBlock(
 
       throwIfStopped();
 
-      final DataGas newExcessDataGas = computeExcessDataGas(transactionResults, newProtocolSpec);
+      DataGas newExcessDataGas = null;
+      Long newDataGasUsed = null;
+      if (newProtocolSpec.getFeeMarket().implementsDataFee()) {
+        final var gasCalculator = newProtocolSpec.getGasCalculator();
+        newExcessDataGas =
+            DataGas.of(
+                gasCalculator.computeExcessDataGas(
+                    // casting parent excess data gas to long since for the moment it should be well
+                    // below that limit
+                    parentHeader.getExcessDataGas().map(DataGas::toLong).orElse(0L),
+                    parentHeader.getDataGasUsed().orElse(0L)));
+
+        final int newBlobsCount =
+            transactionResults.getTransactionsByType(TransactionType.BLOB).stream()
+                .map(tx -> tx.getVersionedHashes().orElseThrow())
+                .mapToInt(List::size)
+                .sum();
+
+        newDataGasUsed = gasCalculator.dataGasUsed(newBlobsCount);
+      }
 
```

Comment:
Think this is missing unit tests?

Reply:
It is, I have added a skeleton of one, but completing it turned into a rabbit hole, so I will leave the test ignored for now, and complete it once this branch is merged to main.